lib/FieldLocation/FieldLocationFinal.js

```diff
@@ -15,26 +16,31 @@ import {
   TextField,
 } from '@folio/stripes/components';
 
+import { FieldSelectFinal } from '../FieldSelect';
 import { locationByPropManifest } from '../manifests';
 import { validateRequired } from '../utils';
+import { fieldSelectOptionsShape } from '../shapes';
 
 const FieldLocationFinal = ({
-  locationId,
   labelId,
+  locationId,
+  locationLookupLabel,
+  locationOptions,
   mutator,
   name,
   onChange,
   required,
 }) => {
-  const [selectedLocation, setSelectedLocation] = useState({});
+  const [selectedLocation, setSelectedLocation] = useState();
+  const isPoLineLocation = Boolean(find(locationOptions, { 'value': locationId }));
```

Comment:
this logic looks weird. Why would Field now anything about po line or no?

Reply:
ok, since no response, maybe I can elaborate more on that. Since it's a common component I would expect more generic approach, something like optional array prop with location ids, so list could be pre-populated with them.
Also, it's always be a select dropdown due to new requirements, no need in ifs in render for textfield

Reply:
Sorry for slow reply. 
Firstly, I agree that a common component shouldn't know anything about poLine. The idea of `isPoLineLocation` is to check `locationId` in `locationOptions`. It can be renamed.
Seconly, It's not clear from new requirements that the location field is always `Select`. Story says -  if user clicks "Assign a different location", location is added and can be cleared by clicking the 'X'. So the approach is to show a dropdown list with locations. But if user wants to assign a different one from `LocationLookup` it's would be shown as a disabled text field with 'clear' button. As I can see [Select ](https://ux.folio.org/storybook/?selectedKind=Select&selectedStory=Basic%20Usage&full=0&addons=1&stories=1&panelRight=0&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel)doesn't have prop `endControl` to render 'X' button. It has only [`TextField`](https://ux.folio.org/storybook/?selectedKind=TextField&selectedStory=Basic%20Usage&full=0&addons=1&stories=1&panelRight=0&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel).
Let's discuss it